server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -570,7 +570,7 @@ protected Node(final Environment initialEnvironment,
             final Transport transport = networkModule.getTransportSupplier().get();
             Set<String> taskHeaders = Stream.concat(
                 pluginsService.filterPlugins(ActionPlugin.class).stream().flatMap(p -> p.getTaskHeaders().stream()),
-                Stream.of(Task.X_OPAQUE_ID)
+                Stream.of(Task.X_OPAQUE_ID, Task.TRACE_PARENT)
```

Comment:
Does this make sure the header gets propagated to other downstream ES nodes?

Reply:
Afaik this makes sure the headers go over the transport wire to other nodes.

Reply:
Even though it would technically render the `traceparent` header [invalid](https://www.w3.org/TR/trace-context/#parent-id), we may want to zero out the `parent-id` part of the header when sending to downstream nodes.

```
Kibana 
  |
traceparent: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01
  |
  v
ES Node1
  |
traceparent: 00-0af7651916cd43dd8448eb211c80319c-0000000000000000-01
  |
  v
ES Node 2
```

That is because ES is not creating spans currently and thus there's no parent.
Based on the outcome of this discussion (https://github.com/elastic/elasticsearch/pull/74210#discussion_r653280564), we don't need that part of the `traceparent` header anyway.

Reply:
Removed `transaction.id`. 

I personally prefer not zeroing out and treat ES as pass-through for now given this piece of information is no longer used.